examples/linux/io_epoll_test.cpp

```diff
@@ -87,6 +117,115 @@ int main() {
   } catch (const std::exception& ex) {
     std::printf("error: %s\n", ex.what());
   }
+
+  auto pipe_bench = [](auto& rPipeRef, auto& buffer, auto scheduler, int seconds, auto& data, auto& reps, auto& offset) {
+    return defer(
+      [&, scheduler, seconds] {
+        return
+          transform_done(
+            stop_when(
```

Comment:
I was toying around with migrating the `stop_when()` call out of the `pipe_bench` algorithm and instead building a `stop_after(s, dur)` algorithm that produced `transform_done(stop_when(s, schedule_after(d)))`.

Then the pipe_bench algorithm could just be `repeat_effect(...)`.
And the call site would be `stop_after(pipe_bench(), 3s)`.